### PR TITLE
RSDK-7328: Change DigitalIntrruptByName to return a  DigitalInterrupt and an error

### PR DESCRIPTION
**Pull request by martha-johnston**

### CODE REVIEW

File: components/board/genericlinux/board.go [hunk 1]
```diff
@@ -407,24 +407,23 @@ func (b *Board) AnalogByName(name string) (board.Analog, error) {
...
-		return interrupt.interrupt, true
+		return interrupt.interrupt, nil
 	}
 
 	// Otherwise, the name is not something we recognize yet. If it appears to be a GPIO pin, we'll
 	// remove its GPIO capabilities and turn it into a digital interrupt.
 	gpio, ok := b.gpios[name]
 	if !ok {
-		return nil, false
+		return nil, errors.Errorf("cant find GPIO (%s)", name)
```
Comment (randhid) [inline components/board/genericlinux/board.go:423]: use `fmt.Errorf` instead.

File: components/board/hat/pca9685/pca9685.go [hunk 2]
```diff
@@ -273,8 +273,8 @@ func (pca *PCA9685) AnalogByName(name string) (board.Analog, error) {
 }
 
 // DigitalInterruptByName returns the interrupt by the given name if it exists.
-func (pca *PCA9685) DigitalInterruptByName(name string) (board.DigitalInterrupt, bool) {
-	return nil, false
+func (pca *PCA9685) DigitalInterruptByName(name string) (board.DigitalInterrupt, error) {
+	return nil, nil
```
Comment (randhid) [inline components/board/hat/pca9685/pca9685.go:277]: return an error telling the user this is not implemented.

File: components/board/numato/board.go [hunk 3]
```diff
@@ -246,8 +246,8 @@ func (b *numatoBoard) AnalogByName(name string) (board.Analog, error) {
 }
 
 // DigitalInterruptByName returns a digital interrupt by name.
-func (b *numatoBoard) DigitalInterruptByName(name string) (board.DigitalInterrupt, bool) {
-	return nil, false
+func (b *numatoBoard) DigitalInterruptByName(name string) (board.DigitalInterrupt, error) {
+	return nil, nil
```
Comment (randhid) [inline components/board/numato/board.go:250]: same here, return nil with an unimplemented, it's what's expected with go.

File: components/board/pi/impl/board.go [hunk 4]
```diff
@@ -679,36 +679,36 @@ func (pi *piPigpio) AnalogByName(name string) (board.Analog, error) {
...
 			})
 			if err != nil {
-				return nil, false
+				return nil, err
 			}
 			if result := C.setupInterrupt(C.int(bcom)); result != 0 {
 				err := picommon.ConvertErrorCodeToMessage(int(result), "error")
-				pi.logger.Errorf("Unable to set up interrupt on pin %s: %s", name, err)
-				return nil, false
+				err1 := errors.Errorf("Unable to set up interrupt on pin %s: %s", name, err)
```
Comment (randhid) [inline components/board/pi/impl/board.go:702]: You can join the error statements, I think you can use fmt for the one you're making and invoke it directly in the input to errors.Join()
https://pkg.go.dev/errors#Join

**Review by randhid — APPROVED** [review]
I have a few minor error nits, please address them and feel free to merge!

File: components/board/genericlinux/board.go (hunk 1 above)
Comment (randhid) [reply components/board/genericlinux/board.go:423]: the last file you used Errorsf in didn't have the fmt package pulled in. (the pi driver)
genericlinux does, so let's not use the public archive errors package.

File: components/board/pi/impl/board.go (hunk 4 above)
Comment (martha-johnston) [reply components/board/pi/impl/board.go:702]:
```suggestion
				return nil, errors.Errorf("Unable to set up interrupt on pin %s: %s", name, err)
```

Comment (martha-johnston) [reply components/board/pi/impl/board.go:702]: what about just returning this directly? not sure why I didn't do that

Comment (randhid) [reply components/board/pi/impl/board.go:702]: great

**Comment by martha-johnston** [comment]
@randhid do you have any idea why this pi test might be failing? I can't reproduce locally

**Comment by martha-johnston** [comment]
just as an update, this test is passing on my pi so not sure why it's failing here. will keep debugging

**Pull request closed without merging**